### ISSUE TITLE
CentOS 6.2 : Connecting to AD with SSL - issues and solutions

### ISSUE THREAD

**Issue by phrend** [piwik/plugin-LoginLdap#95]
I'm currently testing Piwik 2.12.1 on CentOS 6.2, with PHP 5.3.3, and OpenLDAP 2.4.23. (Yes, I'm aware that I'll be needing a newer PHP version soon!) My companies AD requires SSL, so I installed openldap, and php-ldap, which takes care of adding the required params to my php config.

There are a couple of issues that I encountered - some of this may help others, and some are bug reports...

1) I had to add the following to /etc/openldap/ldap.conf

    TLS_CACERT /path/to/my/cert.crt

2) In the LDAP Servers config section, I couldn't find any other way to make PiWik use SSL, so I had to specify my AD server by its full URL (ldaps://foo.bar:3269), rather than it's simple hostname. Based on the docs here: http://php.net/manual/en/function.ldap-connect.php, it looks like using the URL has the side-effect of ignoring the port specified in the Server Port field, so some of the log lines are a little funky, as they still have the param in the Server Port field appended to the end of the Server URL param, at least in some cases, like:

    INFO LoginLdap[2015-04-13 22:23:43] [2a24c] LdapUsers::makeLdapClient: Using LDAP server ldaps://foo.bar:3269:3269

(I did try leaving the Server Port field empty, but that just defaults to appending the default ":389" on the end, so I decided to fill it in with my port anyway.)

3) The red warning banner at the top of my Administration login screen says,
>"The PHP LDAP extension does not appear to be enabled. It is required for this plugin, please install it."

even though my configuration is working. This is annoying, but does not seem to cause any real problems.

Whew - now that that's out of the way, I just need to figure out how to assign some default permissions to all authenticated users!  :smile:

**Comment by diosmosis** [comment]
> The red warning banner at the top of my Administration login screen says,

Many people are reporting this, not sure why it's an issue now. Can the notification box be closed? Did you ever use the plugin w/o the PHP LDAP extension? Maybe the error occurred once and was never removed... Anyway, I've created an issue here: https://github.com/piwik/plugin-LoginLdap/issues/96

> I just need to figure out how to assign some default permissions to all authenticated users!

The "Initial Sites with View Access For New Users" setting can be used for this. Or, you can store access permissions in LDAP (see https://github.com/piwik/plugin-LoginLdap#piwik-access-synchronization ), though it requires modifying the schema of your LDAP server.

**Comment by phrend** [comment]
Thanks @diosmosis ! Yes, I did initially try using the plugin before I had php-ldap installed, so the first time I saw it, it was a valid alert. There is no visible way to ack the alert box (no little x to click, etc.).

Thanks for the info about setting up user access - I'll be digging in to that next!

FWIW, I did try the following:

    <?php
    echo "function exists: " . function_exists('ldap_connect') . "\n";
    ?>

It returns

    function exists: 1
